samples/end-to-end-testing/README.md

```diff
@@ -0,0 +1,119 @@
+
+- infrastructure is deployed correctly
+- it's possible to open an SSH session to `vm-linux-1` using port 22
+- it's possible to ping `vm-linux-2` from `vm-linux-1` SSH session
+
+![End-to-end scenario](assets/scenario.png)
+
+> NOTE: this is a simple scenario to illustrate how to write a basic end-to-end test. We don't recommend having production virtual machines that exposes SSH port over a public IP address.
+
+## Terraform configuration
+
+The Terraform configuration for this scenario can be found in the [src/main.tf](src/main.tf) file. It contains everything to deploy the Azure infrastructure represented on the figure above.
+
+If you are not familiar with creating a Linux virtual machine using Terraform we recommend that you read [this page of the documentation](https://docs.microsoft.com/azure/developer/terraform/create-linux-virtual-machine-with-infrastructure) before.
+
+## End-to-end test
+
+As mentioned in the introduction, the end-to-end test is written in Go language and uses the Terratest framework. It is defined in the [src/test/end2end_test.go](src/test/end2end_test.go) file.
+
+This is the common structure of a Golang test using Terratest:
+
+```Go
+package test
+
+import (
+    "testing"
+
+    "github.com/gruntwork-io/terratest/modules/terraform"
+    test_structure "github.com/gruntwork-io/terratest/modules/test-structure"
+)
+
+func TestEndToEndDeploymentScenario(t *testing.T) {
+    t.Parallel()
+
+    fixtureFolder := "../"
+
+    // User Terratest to deploy the infrastructure
+    test_structure.RunTestStage(t, "setup", func() {
+        terraformOptions := &terraform.Options{
+            // Indicate the directory that contains the Terraform configuration to deploy
+            TerraformDir: fixtureFolder,
+        }
+
+        // Save options for later test stages
+        test_structure.SaveTerraformOptions(t, fixtureFolder, terraformOptions)
+
+        // Triggers the terraform init and terraform apply command
+        terraform.InitAndApply(t, terraformOptions)
+    })
+
+    test_structure.RunTestStage(t, "validate", func() {
+        // run validation checks here
+        terraformOptions := test_structure.LoadTerraformOptions(t, fixtureFolder)
+		publicIpAddress := terraform.Output(t, terraformOptions, "public_ip_address")
+    })
+
+    // When the test is completed, teardown the infrastructure by calling terraform destroy
+    test_structure.RunTestStage(t, "teardown", func() {
+        terraformOptions := test_structure.LoadTerraformOptions(t, fixtureFolder)
+        terraform.Destroy(t, terraformOptions)
+    })
+}
+```
+
+As you can see in the snippet above, the test is composed by three stages:
+
+1. `setup`: this stage is responsible for running Terraform to deploy the configuration.
```

Comment:
A reference to line numbers would be great

Reply:
yes